Constrained_triangulation_3/include/CGAL/Conforming_constrained_Delaunay_triangulation_3.h

```diff
@@ -0,0 +1,3918 @@
+    const auto npos = face_constraint_misses_subfaces_npos;
+    auto i = face_constraint_misses_subfaces_find_first();
+    bool has_missing_subfaces = i != npos;
+    while(i != npos) {
+      const CDT_2& cdt = face_cdt_2[i];
+      for(const auto fh: cdt.finite_face_handles()) {
+        if (false == fh->info().is_outside_the_face &&
+            true == fh->info().missing_subface)
+        {
+          write_2d_triangle(out, fh);
+        }
+      }
+      i = face_constraint_misses_subfaces_find_next(i);
+    }
+    return has_missing_subfaces;
+  }
+
+  /// @{
+  /// remove functions cannot be called
+  void remove(Vertex_handle) = delete;
+  void remove_cluster() = delete;
+  /// @}
+
+protected:
+  Insert_in_conflict_visitor insert_in_conflict_visitor = {this};
+  std::vector<CDT_2> face_cdt_2;
+  bool cdt_2_are_initialized = false;
+  struct Face_edge {
+    Constrained_polyline_id constrained_polyline_id;
+    bool is_reverse = false;
+  };
+  std::vector<std::vector<std::vector<Face_edge>>> face_borders;
+  std::multimap<Constrained_polyline_id, CDT_3_signed_index> constraint_to_faces;
+
+  // boost::dynamic_bitset<> face_constraint_misses_subfaces;
+  // std::size_t face_constraint_misses_subfaces_find_first() const {
+  //   return face_constraint_misses_subfaces.find_first();
+  // }
+  // std::size_t face_constraint_misses_subfaces_find_next(std::size_t pos) const {
+  //   return face_constraint_misses_subfaces.find_next(pos);
+  // }
+  // void face_constraint_misses_subfaces_set(std::size_t pos) {
+  //   face_constraint_misses_subfaces.set(pos);
+  // }
+  // void face_constraint_misses_subfaces_reset(std::size_t pos) {
+  //   face_constraint_misses_subfaces.reset(pos);
+  // }
+  // static inline constexpr std::size_t face_constraint_misses_subfaces_npos = boost::dynamic_bitset<>::npos;
+  // static_assert(false == CGAL::is_nothrow_movable_v<boost::dynamic_bitset<>>);
+  std::vector<bool> face_constraint_misses_subfaces;
+  std::size_t face_constraint_misses_subfaces_find_first(std::size_t pos = 0) const {
+    auto it = std::find(face_constraint_misses_subfaces.begin() + pos, face_constraint_misses_subfaces.end(), true);
+    return it == face_constraint_misses_subfaces.end() ? face_constraint_misses_subfaces_npos
+                                                       : std::distance(face_constraint_misses_subfaces.begin(), it);
+  }
+  std::size_t face_constraint_misses_subfaces_find_next(std::size_t pos) const {
+    return face_constraint_misses_subfaces_find_first(pos + 1);
+  }
+  void face_constraint_misses_subfaces_set(std::size_t pos) {
+    face_constraint_misses_subfaces[pos] = true;
+  }
+  void face_constraint_misses_subfaces_reset(std::size_t pos) {
+    face_constraint_misses_subfaces[pos] = false;
+  }
+  static inline constexpr std::size_t face_constraint_misses_subfaces_npos = boost::dynamic_bitset<>::npos;
+  std::vector<std::size_t> faces_region_numbers;
```

Comment:
I think this should be ` CDT_3_signed_index`    or `int`, because an entry of the vector will later be passed to tghe function  `restore_subface_region()`.